sanic/request.py

```diff
@@ -519,8 +519,11 @@ def url_for(self, view_name, **kwargs):
         :rtype: str
         """
         # Full URL SERVER_NAME can only be handled in app.url_for
-        if "//" in self.app.config.SERVER_NAME:
-            return self.app.url_for(view_name, _external=True, **kwargs)
+        try:
+            if "//" in self.app.config.SERVER_NAME:
+                return self.app.url_for(view_name, _external=True, **kwargs)
+        except AttributeError:
+            pass
```

Comment:
Nevermind my earlier comment; this is OK, although checking in the if instead of try-except could be clearer.

Reply:
> Perhaps you could also add a request.url_for test without SERVER_NAME?

I think there are tests around this already and don't think there is a need to add another one. But Let me check the coverage and see if it's missing. In case if missing, let me add one. 

> although checking in the if instead of try-except could be clearer.

I did not want to do an additional check. If this works, great, we go ahead and return and if it doesn't we just ignore and continue. This way we don't add anything extra in the branching structure.

Reply:
`request.server_name` property actually uses `app.config.get("SERVER_NAME")` and tests that for truthiness. That way `SERVER_NAME=""` behaves the same as if it wasn't defined at all.

Reply:
As for tests, I gather that this wasn't tested for because otherwise #1707 would've been caught in testing. I suppose that only `app.url_for` gets thorough testing and that all existing `request.url_for` tests define `SERVER_NAME` to something.

Reply:
> this wasn't tested for because otherwise #1707 would've been caught in testing

Agreed. I realized that after I wrote the message and hit submit.

Reply:
@Tronic Added an extra test to make sure there is a check when `SERVER_NAME` is not set.